File: algoritimos-e-programacao/estruturas_lineares/imbd/go/filme/filme.go
package filme

type Filme struct {
    codigo, ano int
    nome, diretor string
}

func NovoFilme(codigo int, ano int, nome string, diretor string) *Filme {
    return &Filme{codigo, ano, nome, diretor}
}

func (f *Filme) BuscarNome() string {
    return f.nome
}

func (f *Filme) BuscarAno() string {
    return f.ano
}

File: estrutura-de-dados-2/imdb2/src/main/java/br/com/gabrielgmusskopf/imdb2/Filme.java
package br.com.gabrielgmusskopf.imdb2;

public class Filme extends AudiovisualAbstrato {

    private final double duracao;

    public Filme(String nome, String codigo, int ano, double duracao, Genero genero) {
        super(nome, codigo, ano, genero);
        this.duracao = duracao;
    }

    @Override
    public TipoGravacao getTipoGravacao() {
        return TipoGravacao.FILME;
    }

}
